atomic_reactor/plugins/post_generate_maven_metadata.py

```diff
@@ -56,6 +70,96 @@ def get_nvr_components(self, nvr_requests):
+
+            artifact = {
+                'url': url_request['url'],
+                'checksums': artifact_checksums,
+                'filename': os.path.basename(url_request['url'])
+            }
+
+            if 'source-url' not in url_request:
+                self.no_source_artifacts.append(artifact)
+                msg = f"No source-url found for {url_request['url']}.\n"
+                msg += 'fetch-artifacts-url without source-url is deprecated'
+                self.log.warning(msg)
+                self.log.user_warning(msg)
+                continue
+
+            source_url = url_request['source-url']
+
+            checksums = {algo: url_request[('source-' + algo)] for algo in
+                         hashlib.algorithms_guaranteed
+                         if ('source-' + algo) in url_request}
+
+            if source_url not in self.source_url_to_artifacts:
+                self.source_url_to_artifacts[source_url] = [artifact]
+                # source_url will mostly be gerrit URLs that don't have filename
+                #  in the URL itself so we'll have to get filename from URL response
+                target = os.path.basename(source_url)
+
+                download_queue.append(DownloadRequest(source_url, target, checksums))
+            else:
+                self.source_url_to_artifacts[source_url].append(artifact)
+        return download_queue
+
+    def download_sources(self, download_queue):
+        remote_source_files = []
+        downloads_path = os.path.join(self.workdir, self.DOWNLOAD_DIR)
+
+        session = util.get_retrying_requests_session()
+
+        self.log.debug('%d files to download', len(download_queue))
+
+        koji_config = get_koji(self.workflow)
+        insecure = koji_config.get('insecure_download', False)
+
+        for index, download in enumerate(download_queue):
+            dest_filename = download.dest
+            if not re.fullmatch(r'^[\w\-.]+$', dest_filename):
+                dest_filename = session.get(download.url).headers.get(
+                    "Content-disposition").split("filename=")[1].replace('"', '')
+
+            dest_path = os.path.join(downloads_path, dest_filename)
+            dest_dir = os.path.dirname(dest_path)
+
+            if not os.path.exists(dest_dir):
+                os.makedirs(dest_dir)
+
+            self.log.debug('%d/%d downloading %s', index + 1, len(download_queue),
+                           download.url)
+
+            download_url(url=download.url, dest_dir=dest_dir, insecure=insecure,
+                         session=session, dest_filename=dest_filename,
+                         expected_checksums=download.checksums)
+
+            checksum_type = list(download.checksums.keys())[0]
+
+            remote_source_files.append({
+                'file': dest_path,
+                'metadata': {
+                    'type': KOJI_BTYPE_REMOTE_SOURCE_FILE,
+                    'checksum_type': checksum_type,
```

Comment:
maybe we can have dict of all checksums here as well?

Reply:
I don't think brew doesn't accept checksums as dict :(

Reply:
ic ok then